### PR TITLE
docstore: add a RevisionField option to all providers

### LINKED ISSUE #2154: docstore: make revision field a collection option
When a collection is opened, let the user specify the name of the field that holds the document revision as an option.

I originally thought we could make this a struct tag, but we can't: we need for the entire collection to use the same field, so we can get and set it. As an example of what could go wrong if it were a struct tag, consider saving a document using a struct with a revision tag, then reading it back into a `map[string]interface{}`. How would we know which of the document fields is the revision? We need that information so we can set up the right precondition on subsequent writes.

### PR DESCRIPTION
**Pull request by jba**
Fixes #2154

### CODE REVIEW

File: internal/docstore/dynamodocstore/dynamo.go [hunk 1]
```diff
@@ -207,6 +211,9 @@ func newCollection(db *dyn.DynamoDB, tableName, partitionKey, sortKey string, op
 	if opts == nil {
 		opts = &Options{}
 	}
+	if opts.RevisionField == "" {
+		opts.RevisionField = docstore.RevisionField
```
Comment (shantuo) [inline internal/docstore/dynamodocstore/dynamo.go]: Maybe rename the `docstore.RevisionField` to `docstore.DefaultRevisionField`?

File: internal/docstore/firedocstore/codec.go [hunk 2]
```diff
@@ -142,7 +141,7 @@ func floatval(x float64) *pb.Value { return &pb.Value{ValueType: &pb.Value_Doubl
 ////////////////////////////////////////////////////////////////
 
 // decodeDoc decodes a Firestore document into a driver.Document.
-func decodeDoc(pdoc *pb.Document, ddoc driver.Document, nameField string) error {
+func decodeDoc(pdoc *pb.Document, ddoc driver.Document, nameField, revField string) error {
```
Comment (shantuo) [inline internal/docstore/firedocstore/codec.go]: I suggest wrap `nameField` and `revField` into an options struct.

File: internal/docstore/drivertest/driverbenchmark.go [hunk 3]
```diff
@@ -39,7 +39,6 @@ func RunBenchmarks(b *testing.B, coll *docstore.Collection) {
 	b.Run("BenchmarkActionListGet", func(b *testing.B) {
 		benchmarkActionListGet(100, b, coll)
 	})
-	cleanUpTable(b, coll)
```
[the remark this answers is not in the document]
Comment (jba) [reply internal/docstore/drivertest/driverbenchmark.go]: Restored.

File: internal/docstore/dynamodocstore/dynamo.go (hunk 1 above)
Comment (jba) [reply internal/docstore/dynamodocstore/dynamo.go]: Good idea. Done.

File: internal/docstore/firedocstore/codec.go (hunk 2 above)
Comment (jba) [reply internal/docstore/firedocstore/codec.go]: I see your point. Maybe if we add one more thing that will be worth it.

**Pull request merged**